lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlobFileSystem.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.WindowsAzure.Storage.Blob;
+using NSubstitute;
+using Org.Apache.REEF.IO.FileSystem;
+using Org.Apache.REEF.IO.FileSystem.AzureBlob;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+
+namespace Org.Apache.REEF.IO.Tests
+{
+    /// <summary>
+    /// Test class for unit testing the AzureBlobFileSystem.
+    /// Some methods are currently not unit tested due to the complexity of "faking out"
+    /// the methods. They are instead tested E2E in TestAzureBlobFileSystemE2E
+    /// </summary>
+    [TestClass]
+    public sealed class TestAzureBlobFileSystem
+    {
+        private readonly static Uri FakeUri = new Uri("http://fake.com");
+
+        [TestMethod, ExpectedException(typeof(NotSupportedException))]
+        public void TestCreateNotSupported()
+        {
+            new TestContext().GetAzureFileSystem().Create(FakeUri);
+        }
+
+        [TestMethod, ExpectedException(typeof(NotSupportedException))]
+        public void TestOpenNotSupported()
+        {
+            new TestContext().GetAzureFileSystem().Open(FakeUri);
+        }
+
+        [TestMethod]
+        public void TestDelete()
+        {
+            var testContext = new TestContext();
+            testContext.GetAzureFileSystem().Delete(new Uri(FakeUri, "container/file"));
+            testContext.TestCloudBlob.Received(1).Delete();
+        }
+
+        [TestMethod, ExpectedException(typeof(ArgumentOutOfRangeException))]
+        public void TestDeleteAtContainer()
+        {
+            new TestContext().GetAzureFileSystem().Delete(new Uri(FakeUri, "container"));
+        }
+
```

Comment:
add test for:
Throws exception when underlying call fails except when statuscode is NotFound.